Tools/clinic/clinic.py

```diff
@@ -1226,12 +1246,19 @@ def parser_body(
             limited_capi = False
 
         parsearg: str | None
+        if f.kind in {GETTER, SETTER} and parameters:
+            fail("neither @getter nor @setter can define parameters")
```

Comment:
I think we could give a slightly nicer error message here:

```suggestion
            fail(f"@{f.kind.name.lower()} method cannot define parameters")
```